src/test/java/seedu/address/logic/commands/AddAppointmentCommandTest.java

```diff
@@ -0,0 +1,310 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ClinicAssistant;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyClinicAssistant;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.person.Person;
+import seedu.address.model.timeslots.Timeslot;
+
+public class AddAppointmentCommandTest {
+
+    @Test
+    public void constructor_validAppointment_success() {
+        AddAppointmentCommand addAppointmentCommand = new AddAppointmentCommand(INDEX_FIRST_PERSON,
+                "Health check up", LocalDateTime.of(2021, 1, 1, 12, 0));
+        assert(addAppointmentCommand != null);
+    }
+
+    @Test
+    public void constructor_nullAppointment_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddAppointmentCommand(null,
+                null, null));
+    }
+
+    @Test
+    public void execute_appointmentAcceptedByModel_addSuccessful() {
+        ModelStubAcceptingAppointmentAdded modelStub = new ModelStubAcceptingAppointmentAdded();
+        AddAppointmentCommand addAppointmentCommand = new AddAppointmentCommand(INDEX_FIRST_PERSON,
+                "Health check up", LocalDateTime.of(2024, 1, 20, 12, 0));
+        try {
+            CommandResult commandResult = addAppointmentCommand.execute(modelStub);
+            Appointment newAppointment = new Appointment("Health check up",
```

Comment:
could you change these test cases to now include bryans addition of Doctor as a required field

Reply:
@simbayippy Updated please check